editoast/core_task/src/envs/pathfinding.rs

```diff
@@ -274,11 +278,8 @@ where
         }
     }
 
-    pub(in crate::envs) fn input_train_set(&self, input: &Input) -> TrainSet<Train> {
-        self.input_index
-            .get(input)
-            .expect("Runner::new invariant")
-            .clone()
+    pub(in crate::envs) fn input_train_set(&self, input: &Input) -> Option<TrainSet<Train>> {
+        self.input_index.get(input).map(|trains| trains.clone())
```

Comment:
The following doesn’t work?

```suggestion
        self.input_index.get(input).cloned()
```

Reply:
Nope, it's a dashmap, so the type is [Option](https://doc.rust-lang.org/nightly/core/option/enum.Option.html)<[Ref](https://docs.rs/dashmap/latest/dashmap/mapref/one/struct.Ref.html)<'a, K, V>> and `dashmap::Ref` does not implement `Clone`. Here we clone through `Deref`.

Reply:
Ah, so you mean the following might work then.

```suggestion
        self.input_index.get(input).as_deref().cloned()
```

Reply:
Of course 🤦 https://github.com/OpenRailAssociation/osrd/compare/1174adf8ffc013a771b81de9a4cb3bc10de624f3..47a4db55634067aa88a11b8e535194e0fcf4308f